### DISCUSSION TITLE
Support for GitLab Code Snippets

### DISCUSSION THREAD

**Discussion by 0Styless** [gabrie30/ghorg#331] [Ideas]
Hey,

I would like to ask if there is already a feature to clone GitLab's Code Snippets? Maybe I'm missing it.
If not, it would be a great addition to the already existing function to clone wikis.

https://docs.gitlab.com/ee/user/snippets.html

**Comment by gabrie30** [comment]
Not a feature of ghorg yet but this would be great I'll look into it.

**Comment by han911976** [comment]
For future reference and fun,
I will jot down how I downloaded all the snippets from a self-hosted gitlab here.

1. get all snippet repo url by REST API
find all required url from `http_url_to_repo` 
> [https://docs.gitlab.com/ee/api/snippets.html#list-all-snippets](https://docs.gitlab.com/ee/api/snippets.html#list-all-snippets)

2. forloop in powershell
```powershell
$repo = "https://gitlab.example.com/group1/snippets/99.git,https://gitlab.example.com/group1/snippets/98.git,..."
$repo = $repo.split(",");

foreach ($i in $repo){
  $j = "W:\_code\ghorg\snippet\$($i.split('/')[-1])"
  git clone $i $j
#  cd $j
#  git fetch --all
#  cd ..
  Start-Sleep -m 700
}
```

This should allow quick backup for snippets on the server, but require manual indexing after cloning the snippets, which will be painful if there are a lot of files...

**Comment by gabrie30** [comment]
Been working on some ideas how to do this. Will be similar to the wikis. However I noticed some snippets are not tied to a repo and can be created on the org level instead. Was thinking these can go into a separate folder.

**Reply by gabrie30** [reply]
Also snippets can have the same title so collisions could be common so I'm thinking about prepending the snippet id.

**Comment by gabrie30** [comment]
https://github.com/gabrie30/ghorg/pull/416
